video/out/wayland_common.c

```diff
@@ -290,34 +294,92 @@ static void pointer_handle_button(void *data, struct wl_pointer *wl_pointer,
     }
 }
 
-static void pointer_handle_axis(void *data, struct wl_pointer *wl_pointer,
-                                uint32_t time, uint32_t axis, wl_fixed_t value)
+static void handle_axis_event(struct vo_wayland_state *wl, uint32_t axis, double val)
 {
-    struct vo_wayland_state *wl = data;
-
-    double val = wl_fixed_to_double(value) < 0 ? -1 : 1;
     switch (axis) {
     case WL_POINTER_AXIS_VERTICAL_SCROLL:
-        if (value > 0)
+        if (val > 0)
             mp_input_put_wheel(wl->vo->input_ctx, MP_WHEEL_DOWN | wl->mpmod, +val);
-        if (value < 0)
+        if (val < 0)
             mp_input_put_wheel(wl->vo->input_ctx, MP_WHEEL_UP | wl->mpmod, -val);
         break;
     case WL_POINTER_AXIS_HORIZONTAL_SCROLL:
-        if (value > 0)
+        if (val > 0)
             mp_input_put_wheel(wl->vo->input_ctx, MP_WHEEL_RIGHT | wl->mpmod, +val);
-        if (value < 0)
+        if (val < 0)
             mp_input_put_wheel(wl->vo->input_ctx, MP_WHEEL_LEFT | wl->mpmod, -val);
         break;
     }
 }
 
+static void pointer_handle_axis(void *data, struct wl_pointer *wl_pointer,
+                                uint32_t time, uint32_t axis, wl_fixed_t value)
+{
+    struct vo_wayland_state *wl = data;
+    wl->axis_value += wl_fixed_to_double(value);
+    wl->axis_type = axis;
+}
+
+static void pointer_handle_frame(void *data, struct wl_pointer *wl_pointer)
+{
+    struct vo_wayland_state *wl = data;
+    // Prefer axis_value120 if supported and the axis event is from mouse wheel.
+    if (wl->axis_is_wheel)
```

Comment:
Does it work to change the check here to see `axis_value120` is nonzero? I think this is supposed to be the end of a sequence of events so I don't think it would be possible to say have `axis_value` and `axis_value120` at the same time since the values are always unconditionally being cleared (as they should) after this.

Reply:
Maybe something like

```
double val = 0;

if (wl->axis_value120)
     val = whatever

if (wl->axis_value)
    val = whatever

// stuff
```

Then you could eliminate `axis_is_wheel` from the global state.

Reply:
Ah I see.